card/src/main/java/com/adyen/checkout/card/CardDelegate.kt

```diff
@@ -1,86 +1,42 @@
-) : PaymentMethodDelegateOld {
+interface CardDelegate : PaymentMethodDelegate<CardConfiguration, CardInputData, CardOutputData, CardComponentState> {
 
-    protected val noCvcBrands: Set<CardType> = hashSetOf(CardType.BCMC)
+    val inputData: CardInputData
 
-    abstract fun validateCardNumber(
-        cardNumber: String,
-        enableLuhnCheck: Boolean,
-        isBrandSupported: Boolean
-    ): FieldState<String>
+    val outputDataFlow: Flow<CardOutputData?>
 
-    abstract fun validateExpiryDate(
-        expiryDate: ExpiryDate,
-        expiryDatePolicy: Brand.FieldPolicy?
-    ): FieldState<ExpiryDate>
+    val componentStateFlow: Flow<CardComponentState?>
 
-    abstract fun validateSecurityCode(securityCode: String, cardType: DetectedCardType? = null): FieldState<String>
-    abstract fun validateHolderName(holderName: String): FieldState<String>
-    abstract fun validateSocialSecurityNumber(socialSecurityNumber: String): FieldState<String>
-    abstract fun validateKcpBirthDateOrTaxNumber(kcpBirthDateOrTaxNumber: String): FieldState<String>
-    abstract fun validateKcpCardPassword(kcpCardPassword: String): FieldState<String>
+    val exceptionFlow: Flow<CheckoutException>
 
-    abstract fun validateAddress(
-        addressInputModel: AddressInputModel,
-        addressFormUIState: AddressFormUIState
-    ): AddressOutputData
+    fun initialize(coroutineScope: CoroutineScope)
 
-    abstract fun isCvcHidden(): Boolean
-    abstract fun isSocialSecurityNumberRequired(): Boolean
-    abstract fun isKCPAuthRequired(): Boolean
-    abstract fun requiresInput(): Boolean
-    abstract fun isHolderNameRequired(): Boolean
+    fun isDualBrandedFlow(cardOutputData: CardOutputData): Boolean
 
-    abstract fun getAddressFormUIState(
-        addressConfiguration: AddressConfiguration?,
-        addressVisibility: AddressVisibility
-    ): AddressFormUIState
+    fun isInstallmentsRequired(cardOutputData: CardOutputData): Boolean
 
-    abstract fun isAddressRequired(addressFormUIState: AddressFormUIState): Boolean
+    fun isHolderNameRequired(): Boolean
 
-    abstract fun detectCardType(
-        cardNumber: String,
-        publicKey: String?,
-        coroutineScope: CoroutineScope
-    ): List<DetectedCardType>
+    fun showStorePaymentField(): Boolean
 
-    abstract fun getFundingSource(): String?
+    @StringRes fun getKcpBirthDateOrTaxNumberHint(input: String): Int
 
-    abstract fun getInstallmentOptions(
-        installmentConfiguration: InstallmentConfiguration?,
-        cardType: CardType?,
-        isCardTypeReliable: Boolean
-    ): List<InstallmentModel>
+    fun requiresInput(): Boolean
 
-    abstract fun getSupportedCardTypes(): List<CardType>
-
-    suspend fun fetchPublicKey(): Result<String> {
-        return publicKeyRepository.fetchPublicKey(
-            environment = cardConfiguration.environment,
-            clientKey = cardConfiguration.clientKey
-        )
-    }
+    fun clear()
```

Comment:
Nit: I would stay consistent with the naming as in `ViewModel`s, so this would be `onCleared`